src/core_modules/capture-core/hooks/useAvailableProgramStages.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+import { useMemo } from 'react';
+import log from 'loglevel';
+import { useCommonEnrollmentDomainData } from '../components/Pages/common/EnrollmentOverviewDomain';
+import { errorCreator } from '../../capture-core-utils';
+import type { ProgramStage } from '../metaData';
+import { useProgramFromIndexedDB } from '../utils/cachedDataHooks/useProgramFromIndexedDB';
+
+export const useAvailableProgramStages = (programStage: ProgramStage, teiId: string, enrollmentId: string, programId: string) => {
+    const { error: enrollmentsError, enrollment } = useCommonEnrollmentDomainData(teiId, enrollmentId, programId);
+    const {
+        loading: programLoading,
+        error: programError,
+        program,
+    } = useProgramFromIndexedDB(programId);
+
+    if (enrollmentsError || programError) {
+        log.error(errorCreator('Widget Event Edit could not be loaded')({
+            enrollmentsError,
+            programError,
+        }));
+    }
+    const availableProgramStages = useMemo(() => programStage.allowGenerateNextVisit
+    && !programLoading && program?.programStages
+        ?.reduce((accStage, currentStage) => {
+            if (currentStage.id !== programStage.id) {
```

Comment:
Why do you exclude the currently selected `programStage.id` from the `availableProgramStages`? 
The modal is not shown when the only available stage is the one where an event is completed.

https://user-images.githubusercontent.com/6817073/199670448-e4333fc4-f166-4709-8955-46c05bd10996.mov

Thanks!